daemon/create.go

```diff
@@ -139,6 +139,15 @@ func (daemon *Daemon) VolumeCreate(name, driverName string, opts map[string]stri
 		name = stringid.GenerateNonCryptoID()
 	}
 
+	daemon.volumes.mu.Lock()
```

Comment:
Well we do want it to be created if it doesn't exist, which would be the only case it would actually be created.

The thing with having it return an error when it already exists is it would break named volumes on run.